tutorial.md

```diff
@@ -0,0 +1,1196 @@
+one used above, you can find it [here](https://i.imgur.com/9MQHuZx.png).
+
+## Final Touches
+
+There may be some programs that you don't want xmonad to tile.  The
+classic example here is the [GNU Image Manipulation Program]; it pops up
+all sorts of new windows all the time, and they work best at defined
+sizes.  It makes sense for xmonad to float these kinds of windows by
+default.
+
+This kind of behaviour can be achieved via the `manageHook`, which runs
+when windows are created.  There are several functions to help you match
+on a certain window in [XMonad.ManageHook].  For example, suppose we'd
+want to match on the class name of the application.  With the
+application open, open another terminal and invoke the `xprop` command.
+Then click on the application that you would like to know the properties
+of.  In our case you should see (among other things)
+
+``` shell
+  WM_CLASS(STRING) = "gimp", "Gimp"
+```
+
+The second string in `WM_CLASS` is the class name, which we can access
+with `className` from [XMonad.ManageHook].  The first one is usually
+called the instance name and is matched-on via `appName` from the same
+module.
+
+Let's use the class name for now.  We can tell all windows with that
+class name to float by defining the following manageHook:
+
+``` haskell
+  myManageHook = (className =? "Gimp" --> doFloat)
+```
+
+Say we also want to float all dialogs.  This is easy with the `isDialog`
+function from [XMonad.Hooks.ManageHelpers] (which you should import) and
+a little modification to the `myManageHook` function:
+
+``` haskell
+  myManageHook :: ManageHook
+  myManageHook = composeAll
+      [ className =? "Gimp" --> doFloat
+      , isDialog            --> doFloat
+      ]
+```
+
+Now we just need to tell xmonad to actually use our manageHook.  This is
+as easy as overriding the `manageHook` field in `myConfig`.  You can do
+it like this:
+
+``` haskell
+  myConfig = def
+      { modMask    = mod4Mask      -- Rebind Mod to the Super key
+      , layoutHook = myLayout      -- Use custom layouts
+      , manageHook = myManageHook  -- Match on certain windows
+      }
+    `additionalKeysP`
+      [ ("M-S-z", spawn "xscreensaver-command -lock")
+      , ("M-S-=", unGrab *> spawn "scrot -s"        )
+      , ("M-]"  , spawn "firefox"                   )
+      ]
+```
+
+## The Whole Thing
+
+The full `~/.config/xmonad/xmonad.hs`, in all its glory, now looks like
+this:
+
+``` haskell
+  import XMonad
+
+  import XMonad.Hooks.DynamicLog
+  import XMonad.Hooks.ManageDocks
+  import XMonad.Hooks.ManageHelpers
+  import XMonad.Hooks.StatusBar
+  import XMonad.Hooks.StatusBar.PP
+
+  import XMonad.Util.EZConfig
+  import XMonad.Util.Ungrab
+
+  import XMonad.Layout.Magnifier
+  import XMonad.Layout.ThreeColumns
+
+  import XMonad.Hooks.EwmhDesktops
+
+
+  main :: IO ()
+  main = xmonad
+       . ewmhFullscreen
+       . ewmh
+       . withEasySB (statusBarProp "xmobar" (pure myXmobarPP)) defToggleStrutsKey
+       $ myConfig
```

Comment:
Also a nitpick, I think you can mention in comment that users can use "xmobar ~/path/to/config" so that it is clearer.

Reply:
Also, if they don't go and read the source code, they may though, oh, only xmobar should be here.

Reply:
> Also a nitpick, I think you can mention in comment that users can use "xmobar ~/path/to/config" so that it is clearer. Some users might just use "xmobar" and then ask question about why it does not work.

xmobar reads from `~/.xmobarrc` by default, and this is the path that we tell people to install the config in.  We could put it down to user error if they put it somewhere else, but perhaps you're right and we should mention this.

> Also, if they don't go and read the source code, they may say, oh, only `xmobar` should be here. But actually it can be a command line.

I personally don't know of another bar that fascilitates property-based logging by default, so that assumption may or may not be accurate :)  I think the fact that it's a string and this is configurable should be more-or-less obvious; otherwise it would surely be hardcoded!

Reply:
Yeah, I mean `xmobar` used with `command line options`. I agree with what you said. So it would be OK to mention a bit of it. So do you intend to include a `user error` part in this PR, or in later PR?

Reply:
I added a little something

Reply:
Thanks, it looks nice.